### PR TITLE
Add breakParent support for willBreak

### LINKED ISSUE #615: End of line comments moved with last argument expansion
```js
runtimeAgent.getProperties(
  objectId,
  false, // ownProperties
  false, // accessorPropertiesOnly
  false, // generatePreview
  (error, properties, internalProperties) => {
    return 1
  },
);
```

turns into

```js
runtimeAgent.getProperties(objectId, false, false, false, ( // ownProperties // accessorPropertiesOnly // generatePreview
  error,
  properties,
  internalProperties
) => {
  return 1;
});
```

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22runtimeAgent.getProperties(%5Cn%20%20objectId%2C%5Cn%20%20false%2C%20%2F%2F%20ownProperties%5Cn%20%20false%2C%20%2F%2F%20accessorPropertiesOnly%5Cn%20%20false%2C%20%2F%2F%20generatePreview%5Cn%20%20(error%2C%20properties%2C%20internalProperties)%20%3D%3E%20%7B%5Cn%5Ctreturn%201%5Cn%20%20%7D%2C%5Cn)%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

which then turns into

```js
runtimeAgent.getProperties(objectId, false, false, false, (
  // ownProperties // accessorPropertiesOnly // generatePreview
  error,
  properties,
  internalProperties
) => {
  return 1;
});
```

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22runtimeAgent.getProperties(objectId%2C%20false%2C%20false%2C%20false%2C%20(%20%2F%2F%20ownProperties%20%2F%2F%20accessorPropertiesOnly%20%2F%2F%20generatePreview%5Cn%20%20error%2C%5Cn%20%20properties%2C%5Cn%20%20internalProperties%5Cn)%20%3D%3E%20%7B%5Cn%20%20return%201%3B%5Cn%7D)%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by vjeux**
Fixes #615

### CODE REVIEW

File: tests/last_argument_expansion/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -20,17 +20,16 @@ export default function searchUsers(action$) {
     .ofType(ActionTypes.SEARCHED_USERS)
     .map(action => action.payload.query)
     .filter(q => !!q)
-    .switchMap(q =>
```
Comment (jlongster) [inline tests/last_argument_expansion/__snapshots__/jsfmt.spec.js.snap]: Is this right? This looks like a regression that we fixed before

Comment (vjeux) [inline tests/last_argument_expansion/__snapshots__/jsfmt.spec.js.snap]: I don't know what it should look like. Right now it alternates depending on the context. See https://github.com/jlongster/prettier/issues/625 for another example.

**Comment by jlongster** [comment]
Hm, interesting, I think this is right actually. I wasn't sure about it at first because this will find any `breakParent` even though it won't be propagated through `conditionalGroup`s, but I suppose that happens with all other things that will propagate breaks anyway. I'd still like take a closer look before merging to think about it

**Comment by vjeux** [comment]
I'd love to get this committed as it sounds reasonable and fixes a bunch of the unstable issues we have on fb codebase.

**Comment by jlongster** [comment]
Just rebased this and I'll merge it once green. I'm not sure what to think of thinks like `conditionalGroup` because we'll return true if a hard line or `breakParent` exists anywhere in there even though it might not be printed. We haven't run into an issue with that yet.

**Comment by vjeux** [comment]
(You need to re-record the snapshot tests)

**Pull request merged**